zephyr/include/rtos/interrupt.h

```diff
@@ -0,0 +1,105 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * TODO: IMX support, today IRQ is supported via SOF xtos API.
+ */
+
+#ifndef __ZEPHYR_RTOS_INTERRUPT_H__
+#define __ZEPHYR_RTOS_INTERRUPT_H__
+
+#if defined(CONFIG_IMX)
+/* imx currently has no IRQ driver in Zephyr so we force to xtos IRQ */
+#include "../../../xtos/include/rtos/interrupt.h"
+#else
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <stdint.h>
+
+/* to be removed once driver upstream in Zephyr */
+#define __SOF_DRIVERS_INTERRUPT_H__
+#include <platform/drivers/interrupt.h>
+
+static inline int interrupt_register(uint32_t irq, void(*handler)(void *arg), void *arg)
```

Comment:
isn't it the same code as in wrapper.c for IMX? Why do we need an additional copy?

Reply:
The wrapper will go away over time, it may also split into smaller per feature files.

Reply:
but why is your PR keeping it in wrapper.c for IMX? It seems the same, so they can use the version in the header too?